src/Core/Addon/Module/ModuleRepository.php

```diff
@@ -192,14 +192,27 @@ public function clearCache()
      *
      * @param string $name The technical module name to instanciate
      *
-     * @return \Module|null Instance of legacy Module, if valid
+     * @return LegacyModule|null Instance of legacy Module, if valid
      */
     public function getInstanceByName($name)
```

Comment:
this was wrong because developer mostly didn't understood what is Adapter strategy ;'(